pkg/controller/controller_instance.go

```diff
@@ -2381,6 +2381,40 @@ func (c *controller) prepareDeprovisionRequest(instance *v1beta1.ServiceInstance
 			return nil, nil, stderrors.New("InProgressProperties must be set when there is an operation or orphan mitigation in progress")
 		}
 		rh.inProgressProperties = instance.Status.InProgressProperties
+	} else if instance.Status.CurrentOperation == "" && instance.Status.ProvisionStatus != v1beta1.ServiceInstanceProvisionStatusProvisioned {
+		// terminal provisioing failure
+		// we don't have neither ExternalProperties nor InProgressProperties in Status anymore, so we have to build tme
+		if instance.Spec.ClusterServiceClassSpecified() {
+			servicePlan, err := c.clusterServicePlanLister.Get(instance.Spec.ClusterServicePlanRef.Name)
+			if nil != err {
```

Comment:
nit: `err != nil` is more common in Go, unlike C++ :)

Reply:
copy paste :)